llama_index/query_engine/citation_query_engine.py

```diff
@@ -193,9 +197,14 @@ def _create_citation_nodes(self, nodes: List[NodeWithScore]) -> List[NodeWithSco
         new_nodes: List[NodeWithScore] = []
         for node in nodes:
             text_chunks = self.text_splitter.split_text(node.node.get_content())
```

Comment:
Actually I think ideally, you would just have to set `metadata_mode` in `get_content` to include metadata

i.e. `get_content(metadata_mode=MetadataModel.ALL)` or `get_content(metadata_mode=MetadataModel.LLM)`